File: Java/Intro/src/For.java
public class For {
    public static void main(String[] args) {
        // for = loop
        // de unde incepem?; pana unde mergem?; pasul de parcurgere;
        // problema: printam 101 dalmatieni
        for (int i = 1; i<=101; i++) {
            System.out.println("Dalmatianul cu numarul " + i);
        }

        // for ne ajuta si sa parcurgem un array prin intermediul indexului
        int[] numere = {3, 7, 5, 9};
        for (int i = 0; i < numere.length; i++) {
            System.out.println("Elementul cu indexul " + i + " are valoarea " + numere[i]);
        }
        // parcurgerea din 2 in 2:
        int[] numere2 = {3, 7, 5, 9, 4, 8, 6};
        for (int i = 0; i < numere2.length; i+=2) {
            System.out.println("Elementul cu indexul " + i + " are valoarea " + numere2[i]);
        }

        // for each - parcurge toate elementele din array si ajunge direct la valoare
        for (int numar : numere2) {
            System.out.println("Numarul este: " + numar);
        }
        // for (String culoare : culori)
        String[] culori = {"alb","rosu","galben"};
        for (String culoare : culori) {
            System.out.println("Culoarea actuala este " + culoare);
        }

        // suma numerelor din array:
        int s = 0;
        for (int numar : numere2) {
            s = s + numar;
        }
        System.out.println("Suma este: " + s);
    }
}

File: Python/Intro/for.py
# dalmatienii
for i in range(1,102):
    print(f'Dalmatianul cu numarul {i}')

# din 2 in 2
for i in range(1,102,2):
    print(f'Dalmatianul cu numarul {i}')

numere = [3, 7, 5, 9, 4, 8, 6]
# parcurgere lista cu for prin intermediul indexului:
for i in range(0, len(numere)):
    print(f'Indexul curent este {i}')
    print(f'Numarul curent este {numere[i]}')

# for each:
for numar in numere:
    print(f'Numarul curent este {numar}')
s = 0
for numar in numere:
    print(f'Numarul curent este {numar}')
    s = s + numar
print(f'Suma numerelor este {s}')

# de cate ori apare 3 in [3, 2, 3, 5, 3]?
lista_numere = [3, 2, 3, 5, 3, 4, 3, 3, 4, 3]
cate = 0
for numar in lista_numere:
    if (numar == 3):
        cate+=1
print(f'Cifra 3 apare in sirul nostru de {cate} ori.')
